packages/api/cms-api/src/dam/files/files.resolver.ts

```diff
@@ -108,6 +112,46 @@ export function createFilesResolver({ File, Scope: PassedScope }: { File: Type<F
             return this.filesService.moveBatch(files, targetFolder);
         }
 
+        @Query(() => [File])
+        async getAllFilesUsedOnPage(@Args("pageTreeNodeId", { type: () => ID }) pageTreeNodeId: string): Promise<FileInterface[]> {
+            const node = await this.pageTreeService.createReadApi().getNode(pageTreeNodeId);
+
+            if (node === null) {
+                throw new Error("PageTreeNode not found");
+            }
+
+            const document = await this.pageTreeService.getActiveAttachedDocument(node.id, node.documentType);
+
+            if (document === null) {
+                return [];
+            }
+
+            await this.dependenciesService.refreshViews();
+            const dependencies = await this.dependenciesService.getDependencies(
+                { entityName: document.type, id: document.documentId },
+                { targetEntityName: FILE_ENTITY },
+            );
+
+            const fileIds = dependencies.map((dependency) => dependency.targetId);
+            return this.filesService.findMultipleByIds(fileIds);
+        }
+
+        @Mutation(() => CopyFilesResponse)
+        @SkipBuild()
+        async copyFilesToScope(
+            @Args("fileIds", { type: () => [ID] }) fileIds: string[],
+            @Args("targetScope", { type: () => Scope }) targetScope: typeof Scope,
+            @Args("targetFolderId", {
+                type: () => ID,
+                nullable: true,
+                description:
+                    "You can set this argument to use the same inbox folder for multiple consecutive copy operations. Keep it empty for the first copy operation and get the ID from the response.",
+            })
+            targetFolderId?: string,
+        ): Promise<CopyFilesResponseInterface> {
+            return this.filesService.copyFilesToScope({ fileIds, targetScope, targetFolderId });
```

Comment:
you need to check permissions:

- if the arg is n amed targetScope (not scope) the ScopeGuard doesn't check it
- you need to check if targetFolderId scope matches targetScope (or user has permission to targetFolderId scope)
- And finally you need permissions for source files

Reply:
I added checks to check:
- if the user has access to all the file scopes
- if the user has access to the target scope
- if the target folder is in the target scope